### PR TITLE
Support for Mic selection

### LINKED ISSUE #2799: Add preference to change audio/microphone devices that are used by Hubs
**Is your feature request related to a problem? Please describe.**
Right now, the only way to change the microphone and audio output devices that I am using with Hubs is to refresh the page and change them as part of the room entry flow. We should have a section in the preferences panel that allows us to change our microphone and audio out so that we can narrow down issues with devices being mis-selected without needing to refresh, reload, and re-enter the room, or allow us to change devices without disconnecting entirely.

**Describe the solution you'd like**
A section in the preferences menu where I can see which devices I am currently using, and change them. This will likely need a new browser permission prompt, but that seems acceptable to me if I don't need to refresh the page, especially since refreshing the page doesn't always fix my issue. This could also live in the mute/unmute section of the UI to make it more discoverable.

**Additional context**
I was playing with https://janus.conf.meetecho.com/devicetest.html which seems to indicate that this is technically possible without a page reload.

I am not totally clear on whether this would be needed at this time with VR headsets or mobile phones, since those feel less likely to have as many potential audio in/output settings as a desktop PC would.

### PR DESCRIPTION
**Pull request by keianhzo**
Closes #2799 This PR adds support for Mic selection from preferences.

### CODE REVIEW

**Comment by robertlong** [comment]
@keianhzo could you resolve these merge conflicts? I would just get rid of the changes to the localization files. We don't use `sync-locales.js` anymore. The `preferences-screen.js` file was changed significantly to add localization support. I wasn't super confident in fixing the merge conflicts. Let me know if you'd like to pair on that.

**Comment by keianhzo** [comment]
@robertlong The conflicts should be fixed.

File: src/utils/media-devices-manager.js [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+      const constraints = { audio: { deviceId: { exact: [deviceId] } } };
+      const result = await this.fetchAudioTrack(constraints);
+      await this.setupNewMediaStream();
+      NAF.connection.adapter.enableMicrophone(true);
+
+      return result;
+    } else if (this.isMicDeviceSelected) {
+      const audioSystem = this._scene.systems["hubs-systems"].audioSystem;
+      audioSystem.removeStreamFromOutboundAudio("microphone");
+      this.audioTrack?.stop();
```
[the remark this answers is not in the document]
Comment (keianhzo) [reply src/utils/media-devices-manager.js:102]: Yep! that's useful, I added some time ago. Makes code more readable and compact.

File: src/App.js [hunk 2]
```diff
@@ -6,5 +6,6 @@ export class App {
     this.scene = null;
     this.store = new Store();
     this.mediaSearchStore = new MediaSearchStore();
+    this.mediaDevicesManager = null;
```
[the remark this answers is not in the document]
Comment (keianhzo) [reply src/App.js:9]: Sorry, I run out of time I'll do it as a follow-up when the new redesign lands.

**Comment by keianhzo** [comment]
@robertlong This should be ready to merge. I've removed the translations. I run out of time for the refactor, I'll do it as a follow-up when this lands.

**Pull request merged**